src/averageAges.js

```diff
@@ -55,7 +66,16 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const allWomenAge = people
+    .filter(person => onlyWithSon
+      ? people.some(women => women.name === person.mother)
+      && person.sex === 'm'
+      : people.some(women => women.name === person.mother))
```

Comment:
maybe it`s better to create a variable for women with child

Reply:
Fixed.